packages/web/components/navbar-osmo-price.tsx

```diff
@@ -34,19 +33,15 @@ const NavbarOsmoPrice = observer(() => {
     onClose: onCloseFiatOnrampSelection,
   } = useDisclosure();
 
-  const { nativeBalances } = assetsStore;
-
   const { chainId } = chainStore.osmosis;
   const wallet = accountStore.getWallet(chainId);
 
-  const osmoCurrency = chainStore.osmosis.stakeCurrency;
-  const osmoPrice = priceStore.calculatePrice(
-    new CoinPretty(
-      chainStore.osmosis.stakeCurrency,
-      DecUtils.getTenExponentNInPrecisionRange(
-        chainStore.osmosis.stakeCurrency.coinDecimals
-      )
-    )
+  const { data: osmoCurrency } = api.edge.assets.getAsset.useQuery({
+    findMinDenomOrSymbol: "OSMO",
+  });
+  const { data: osmoPrice } = api.edge.assets.getAssetPrice.useQuery(
+    osmoCurrency!,
+    { enabled: Boolean(osmoCurrency) }
```

Comment:
What do you think about creating a single query for all this data, except for the chart data, to minimize the number of requests?

Reply:
My goal with these was to keep the queries single purpose and composable. Are you proposing we add prices to the getAsset(s) procedures?

They will likely be batched so networking overhead shouldn't be a concern.

Reply:
I'm inclined to agree with John here as I think single purpose queries are more scalable and easier to maintain.

Reply:
Also, these particular queries are used individually elsewhere already. I think batching is added to tRPC as to prevent networking from being a concern when composing queries, allowing us to focus on organizing based on the types of data involved.

Reply:
I agree, this applies to queries that aren't dependent on each other. However, in this specific case, we can't batch the query since we must first wait for `getAsset` to complete before we can begin fetching its price